src/integration/java/io/pinecone/integration/controlPlane/pod/CollectionTest.java

```diff
@@ -178,14 +172,96 @@ public void testIndexFromDifferentMetricCollection() throws InterruptedException
+                    "us-east1-gcp",
+                    "us-east4-gcp",
+                    "us-west1-gcp"
+            ));
+            environments.remove(collection.getEnvironment());
+            String mismatchedEnv = environments.get(new Random().nextInt(environments.size()));
+
+            CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().sourceCollection(collection.getName()).environment(mismatchedEnv);
+            CreateIndexRequestSpec spec = new CreateIndexRequestSpec().pod(podSpec);
+            CreateIndexRequest createIndexRequest = new CreateIndexRequest().name(RandomStringBuilder.build("from-coll", 8)).dimension(dimension).metric(IndexMetric.COSINE).spec(spec);
+            pineconeClient.createIndex(createIndexRequest);
+        } catch (PineconeException exception) {
+            logger.info("Exception: " + exception.getMessage());
+            assertTrue(exception.getMessage().contains("Source collection must be in the same environment as the index"));
+        }
+    }
+
+    @Test
+    @Disabled("Bug reported in #global-cps")
+    public void testCreateIndexWithMismatchedDimension() {
+        try {
+            CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().sourceCollection(collection.getName()).environment(collection.getEnvironment());
+            CreateIndexRequestSpec spec = new CreateIndexRequestSpec().pod(podSpec);
+            CreateIndexRequest createIndexRequest = new CreateIndexRequest().name(RandomStringBuilder.build("from-coll", 8)).dimension(dimension + 1).metric(IndexMetric.COSINE).spec(spec);
+            pineconeClient.createIndex(createIndexRequest);
+        } catch (PineconeException exception) {
+            logger.info("Exception: " + exception.getMessage());
+            assertTrue(exception.getMessage().contains("Index and collection must have the same dimension"));
+        }
+    }
+
+    @Test
+    public void testCreateCollectionFromNotReadyIndex() throws InterruptedException {
+        String notReadyIndexName = RandomStringBuilder.build("from-coll", 8);
+        String newCollectionName = RandomStringBuilder.build("coll-", 8);
+        try {
+            CreateIndexRequestSpecPod specPod = new CreateIndexRequestSpecPod().pods(1).podType("p1.x1").replicas(1).environment(environment);
+            CreateIndexRequestSpec spec = new CreateIndexRequestSpec().pod(specPod);
+            CreateIndexRequest createIndexRequest = new CreateIndexRequest().name(notReadyIndexName).dimension(dimension).metric(IndexMetric.COSINE).spec(spec);
+            pineconeClient.createIndex(createIndexRequest);
+            indexesToCleanUp.add(notReadyIndexName);
+
+            createCollection(pineconeClient, newCollectionName, notReadyIndexName, false);
+        } catch (PineconeException exception) {
+            logger.info("Exception: " + exception.getMessage());
+            assert (exception.getMessage().contains("Source index is not ready"));
```

Comment:
In general using the built-in `assert` mostly maps directly to JUnit `Assert.assertTrue`, it's preferable to avoid using the built-in. 

The reason is b/c it's possible to disable assertion checking in the JVM as a performance optimization. This is unlikely to be disabled in test runs, but you can avoid the chance altogether by just avoiding built-in assert statements.

Reply:
I think @rohanshah18 had also mentioned this to me previously. Going through and updating these.